src/Expr.h

```diff
@@ -700,6 +706,41 @@ class AddExpr final : public BinaryExpr {
     ExprPtr BuildSub(const ExprPtr& op1, const ExprPtr& op2);
 };
 
+// A helper class that enables us to factor some common code.
+class AggrAddDelExpr : public UnaryExpr {
+public:
+    explicit AggrAddDelExpr(ExprTag _tag, ExprPtr _e) : UnaryExpr(_tag, std::move(_e)) {}
+
+    bool IsPure() const override { return false; }
+
+    // Optimization-related:
+    bool IsReduced(Reducer* c) const override { return HasReducedOps(c); }
+    bool HasReducedOps(Reducer* c) const override { return op->HasReducedOps(c); }
+    ExprPtr Reduce(Reducer* c, StmtPtr& red_stmt) override;
```

Comment:
This is implemented for the script-opt case. Should it have a no-op implementation for the non-script-opt case? Same for the duplicate methods below.

Reply:
I'm not following. The usual pattern for `Expr.h` is to include these methods for script optimization (which is all that they're used for). The implementations are in `script_opt/Expr.cc`. The functionality isn't called when running the interpreter.

Reply:
@timwoj I believe this is now the one pending issue. If easier to chat about over Zoom, just let me know.

Reply:
Ok, I see what's happening here now. I can't say that I'm a fan of splitting the implementation of these classes into two places. It just leads to having to dig around for an implementation instead of looking in a single place. It's been like that though, so I guess leave it as it is.

Reply:
FWIW, that split was originally asked for by Robin, per https://github.com/zeek/zeek/pull/1255